src/analysis/wng_path_fwg_position_creator.py

```diff
@@ -0,0 +1,79 @@
+import copy
+from datetime import datetime
+
+import networkx as nx
+
+
+class WNGPathFWGPositionCreator:
+    """
+    Class for creating the positions for plotting the Flood Wave Graph subgraph along a path
+    in the Water Network Graph.
+    """
+    def __init__(self, fwg_subgraph: nx.DiGraph, wng_path: nx.DiGraph):
+        """
+        Constructor.
+        :param x.DiGraph fwg_subgraph: the FWG subgraph along a path in the WNG
+        :param x.DiGraph wng_path: the path in the WNG
+        """
+        self.fwg_subgraph = fwg_subgraph
+        self.reg_numbers = list(wng_path.nodes())
+
+        self.graph_to_plot = None
+        self.positions = None
+
+    def run(self, start_date: str = None, end_date: str = None,
+            plot_all: bool = False):
+        """
+        Run function. Filters the graph between two dates if necessary and creates the positions.
+        :param str start_date: start date of the plot
+        :param str end_date: end date of the plot
+        :param bool plot_all: whether to plot the whole graph or not
+        """
+        if not plot_all:
```

Comment:
Itt a `start_date is None or end_date is None` nem váltja ki a plusz `bool` változót?